ua-snap/cmip6-utils
# Add Visual QC

This PR adds a new `visual_qc.ipynb` notebook, an `indicators/shp/` directory containing a coastlines shapefile, and an updated `environment.yml`. This is a companion [PR #9](https://github.com/ua-snap/prefect/pull/9) in the [add_viz_qc branch](https://github.com/ua-snap/prefect/tree/add_viz_qc) of the prefect repo, which contains a new `indicator_functions.visual_qc_nb()` task/function.

Both `papermill` and `geopandas` packages were added to the `cmip6-utils` conda environment in order to execute the QC notebook. You should therefore update the `cmip6-utils` env before testing this PR by using the new `environment.yml` mentioned above. You might even need to delete the original env first and rebuild from scratch.

**To test:**
- update your `cmip6-utils` env using the new `environment.yml` mentioned above
- confirm you are on the `add_viz_qc` branch of the prefect repo and start the prefect server
- run `python generate_indicators.py` as usual
- create a new flow run using the `visual-qc` branch of the cmip6-utils repo
- confirm the new `visual_qc_nb()` task runs successfully
- check the `output/qc/...` directory for `visual_qc_out.ipynb` and `visual_qc_out.html`
- copy these outputs to your local machine somehow. I used `scp` from my local like so:
```
scp jdpaul3@chinook04.alaska.edu:/import/beegfs/CMIP6/jdpaul3/scratch/output/qc/visual_qc_out.html /Users/joshpaul/Downloads
```
- review the outputs
